app/src/main/java/com/example/criminalintentapp/presentation/RegisterActivity.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.criminalintentapp.presentation
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.lifecycle.ViewModelProvider
+import com.example.criminalintentapp.databinding.ActivityRegisterBinding
+
+class RegisterActivity : AppCompatActivity() {
+
+    private val authenticationViewModel: AuthenticationViewModel by lazy {
+        ViewModelProvider(this).get(AuthenticationViewModel::class.java)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val binding = ActivityRegisterBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        authenticationViewModel.userRegisterLiveData.observe(this) { user ->
+            if (user != null) {
+                Toast.makeText(this, "Registered successfully", Toast.LENGTH_SHORT).show()
+                onBackPressed()
+            }
+        }
+
+        authenticationViewModel.failureLiveData.observe(this) { message ->
+            Toast.makeText(this, message, Toast.LENGTH_LONG)
+                .show()
+        }
+
+        setOnClickListeners(binding)
+    }
+
+    private fun setOnClickListeners(binding: ActivityRegisterBinding) {
+        binding.tvSignIn.setOnClickListener {
+            onBackPressed()
+        }
+
+        binding.btnSignUp.setOnClickListener {
+            if (binding.etSignUpEmail.text.isEmpty() && binding.etSignUpPassword.text.isEmpty()) {
```

Comment:
This check could be done inside the viewModel. You can create a function that returns true if all conditions are met. This function will take the mail & password texts as arguments and return a boolean. This way you can also avoid duplicate code since you do this check both in LoginActivity and RegisterActivity

Reply:
Also, lets make sure that the password is at least 6 characters long